common/test/acceptance/tests/studio/test_studio_settings.py

```diff
@@ -571,3 +572,43 @@ def test_special_exams_menu_a11y(self):
             include=['section.edit-settings-timed-examination']
         )
         self.course_outline.a11y_audit.check_for_accessibility_errors()
+
+
+class StudioSettingsImageUploadTest(StudioCourseTest):
+    """
+    Class to test course settings image uploads.
+    """
+    def setUp(self):  # pylint: disable=arguments-differ
+        super(StudioSettingsImageUploadTest, self).setUp()
+        self.settings_page = SettingsPage(self.browser, self.course_info['org'], self.course_info['number'],
+                                          self.course_info['run'])
+
+    @flaky(max_runs=3, min_passes=2)      # TODO fix this SOL-1776
+    def test_upload_course_card_image(self):
+        self.settings_page.visit()
+        self.settings_page.wait_for_page()
```

Comment:
Hi, yes.... both `.visit()` and `.wait_for_page()` poll until `.is_browser_on_page` evaluates to `True`.

If you can't seem to get past what is making them flaky, @Muddasser or I can spend time on it in the next couple of days to help you pinpoint the problem. It could be a learning opportunity on patterns.... LMK.